### DISCUSSION TITLE
Cart Transform updateOperation is moving line items to bottom of cart

### DISCUSSION THREAD

**Discussion by PhilipBeauford** [Shopify/function-examples#403] [Feedback: Cart Transform API]
I have a related issue regarding the updateOperation.
When I update a price down to zero the line item is then moved to the bottom of the list of cart items, instead of staying in the order it was added to the cart.

My client has a current legacy bundle builder that uses parent products with hidden price fields, and for now the bundle parent product price needs to be hidden as a work-around while we update the bundle logic for extensibility. I'm able to change price to 0 and 'hide' the price field by removing the 'free' label but the parent products are being pushed to the bottom of the item list, negating the use case.

Please advise if the line items can be adjusted to not be relocated to bottom of list after price updates or when updated to $0.

**Comment by loclovepop** [comment]
any solutions @PhilipBeauford ?

**Comment by andrewchen5678** [comment]
cart transform function in general causes bugs where cart line items become out of order randomly after it is applied :(, I had a cart transform function expand operation applied to the shopping cart, and sometimes the expanded items goes to the top of the original item on the shopping cart after visiting the checkout page, and I had to update the shopping cart or visit another page somehow to make it go back to the right order with the expanded item to be below the original item. This makes me not want to use cart transform functions any further because it looks like a half-baked feature.
